Python/bytecodes.c

```diff
@@ -2754,7 +2754,7 @@ dummy_func(
                 GOTO_ERROR(error);
             }
             DECREF_INPUTS();
-            res = _PyObject_CallNoArgsTstate(tstate, enter);
+            res = PyObject_CallNoArgs(enter);
```

Comment:
For my own curiosity, why this seeming pessimization? I thought tstate was being passed along anyways?

Reply:
The problem is that this is a static inline function, and it contains calls to `memcpy` when fully inlined. The compiler assumes that `memcpy` is within 32 bits of the code, which isn't guranteed in the JIT's case.

We *could* emit a trampoline for this one special case, but I figured it would be way easier to use the equivalent non-static-inline version of this function, since these two opcodes aren't particularly hot.

Reply:
We will want to lower `BEFORE_WITH` in the bytecode compiler to something like `LOAD_SPECIAL __enter__; LOAD_SPECIAL __exit__; SWAP 2; CALL` eventually.

So I wouldn't worry about performance here, just do the simplest thing that works.